stack/globals_test.go

```diff
@@ -2796,6 +2796,758 @@ func TestLoadGlobals(t *testing.T) {
+					),
+				},
+			},
+			wantErr: errors.E(hcl.ErrTerramateSchema),
+		},
+		{
+			name:   "globals.map label conflicts with attribute",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/",
+					add: Globals(
+						Str("name", "test"),
+						Map(
+							Labels("name"),
+							Expr("for_each", "[]"),
+							Str("key", "a"),
+							Str("value", "a"),
+						),
+					),
+				},
+			},
+			wantErr: errors.E(globals.ErrRedefined),
+		},
+		{
+			name:   "globals.map missing the value attribute or block",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/",
+					add: Globals(
+						Map(
+							Labels("var"),
+							Expr("for_each", "[]"),
+							Expr("key", "element.new.test"),
+						),
+					),
+				},
+			},
+			wantErr: errors.E(hcl.ErrTerramateSchema),
+		},
+		{
+			name:   "globals.map conflicting value attribute and block - fails",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/",
+					add: Globals(
+						Map(
+							Labels("var"),
+							Expr("for_each", "[]"),
+							Expr("key", "element.new.test"),
+							Str("value", "value"),
+							Value(
+								Str("a", "b"),
+							),
+						),
+					),
+				},
+			},
+			wantErr: errors.E(hcl.ErrTerramateSchema),
+		},
+		{
+			name:   "globals.map validation is recursive",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/",
+					add: Globals(
+						Map(
+							Labels("var"),
+							Expr("for_each", "[]"),
+							Expr("key", "element.new.key"),
+							Expr("value", "element.new.value"),
+							Map(),
+						),
+					),
+				},
+			},
+			wantErr: errors.E(hcl.ErrTerramateSchema),
+		},
+		{
+			name:   "invalid globals.map key",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/",
+					add: Globals(
+						Map(
+							Labels("var"),
+							Expr("for_each", `["a", "b", "c"]`),
+							Expr("key", "something"), // keyword, not a string
+							Str("value", "else"),
+						),
+					),
+				},
+			},
+			wantErr: errors.E(globals.ErrEval),
+		},
+		{
+			name:   "invalid globals.map value",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/",
+					add: Globals(
+						Map(
+							Labels("var"),
+							Expr("for_each", `["a", "b", "c"]`),
+							Str("key", "something"),
+							Expr("value", "else"), // keyword, not a string
+						),
+					),
+				},
+			},
+			wantErr: errors.E(globals.ErrEval),
+		},
+		{
+			name:   "simple globals.map without using element",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/",
+					add: Globals(
+						Map(
+							Labels("var"),
+							Expr("for_each", `["a", "b", "c"]`),
+							Str("key", "something"),
+							Str("value", "else"),
+						),
+					),
+				},
+			},
+			want: map[string]*hclwrite.Block{
+				"/stack": Globals(
+					EvalExpr(t, "var", `{
+						something = "else"
+					}`),
+				),
+			},
+		},
+		{
```

Comment:
maybe now would be a good time to split the tests into different files, like tests focused on map, tests focused on other global behavior, etc. The current test table is quite big. Not a huge thing, just wondering.